### PR TITLE
Logging dead agents during simulation

### PR DESCRIPTION
**Pull request by Rhys-J**
-Added deadAgents mapping parameter to tower struct
-Added UpdateDeadAgents function
-Added DeadAgents function
-Added logs for when agent dies
-Added log for summary of dead agents at end of simulation

# Summary

Added parameter to tower to track dead agents during the simulations.
Added logs for when agent has died and a summary at the end of the simulation.
Added functions to tower.

## Additional Information

Not sure if this is the best way to store and track this information. Please tell me if you have any other suggestions.

Currently works with agentType as int but I understand this is likely to change, so the changes here will have to be modified slightly when this happens.

## Test Plan

Increased the number of simulation days for testing and checked logs over multiple simulations.

### CODE REVIEW

File: pkg/simulation/simAgent.go
Comment (jzzheng22) [inline pkg/simulation/simAgent.go:53]:
```suggestion
			sE.Log("An Agent has died", infra.Fields{"agent_type": agent.AgentType()})
			t.UpdateDeadAgents(agent.AgentType())
			sE.createNewAgent(t, agent.AgentType(), agent.Floor())

```

File: pkg/simulation/simulation.go
Comment (jzzheng22) [inline pkg/simulation/simulation.go:51]:
```suggestion
	sE.Log("Summary of dead agents", infra.Fields{"Agent Type and number that died": t.DeadAgents()})
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
Wait for approval from someone in Team 1 as well, not just me

Couple of minor changes, but otherwise looks good.

Upper case vs lower case in logs matters - changes the order in which things are listed since they're sorted alphabetically

**Review by jzzheng22 — APPROVED** [review]

File: pkg/infra/tower.go
Comment (JaafarRammal) [inline pkg/infra/tower.go:26]: I was considering we could have a struct containing al these "states" that we would like to keep track off and log, and then we pass it to tower and sim and base agent. Everyone can then update accordingly the struct and call something like "state.log" which prints the whole state. What do you think ?

Comment (JaafarRammal) [reply pkg/infra/tower.go:26]: Actually, that's something we can add on the infra team side when we integrate the frontend logging. I think that should be good for now :)

**Review by JaafarRammal — APPROVED** [review]

**Pull request merged**
